flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/BatchTableEnvironment.scala

```diff
@@ -192,10 +193,14 @@ class BatchTableEnvironment(
       name: String,
       f: AggregateFunction[T, ACC])
   : Unit = {
-    implicit val typeInfo: TypeInformation[T] = TypeExtractor
-      .createTypeInfo(f, classOf[AggregateFunction[T, ACC]], f.getClass, 0)
+    implicit val typeInfo: TypeInformation[T] = UserDefinedFunctionUtils
+      .getResultTypeOfAggregateFunction(f)
       .asInstanceOf[TypeInformation[T]]
 
+    implicit val accTypeInfo: TypeInformation[ACC] = UserDefinedFunctionUtils
```

Comment:
UDAGGs are sometimes also shared in libraries. If a UDAGG class is included in a JAR file and loaded into the classpath, a user can hardly tell whether it is implemented in Java or Scala. So, I would not prohibit Scala UDFs in Jave environments. However, functions which are intended to be shared should be implemented with Java types.